### PR TITLE
feature: new and improved thoughts system

### PR DESCRIPTION
**Pull request by Ryos00**
- cats should no longer think about outsiders or dead cats when they're not supposed to
- thoughts are more organized

### CODE REVIEW

File: resources/dicts/thoughts/alive/alive_outside/exile.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            "Wonders why they were exiled",
+            "Rues over their predicament",
+            "Has nightmares about their exile",
+            "Wants a second chance at Clan life",
+            "Fended off their kill from a rogue",
+            "Cried themselves to sleep",
+            "Misses their friends"
+        ],
+        "main_status_constraint": [
+            "exile"
```
Comment (Lixxis) [inline resources/dicts/thoughts/alive/alive_outside/exile.json]:
```suggestion
            "exiled"
```

File: resources/dicts/thoughts/alive/alive_outside/general.json [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            "Got even more lost than they were before",
+            "Worries their Clan thinks they're a runaway traitor",
+            "Finds comfort in their solitude",
+            "Is wandering through the forest",
+            "Is trying to get back to c_n as soon as possible",
+            "Wonders how they will survive without their Clanmates",
+            "Is missing r_c",
+            "Is scared that they will never be found"
+        ],
+        "main_status_constraint": [
```
Comment (Lixxis) [inline resources/dicts/thoughts/alive/alive_outside/general.json:53]: Is the elder not allowed to think this?
The elder is missing in the status list - and if they are implemented like the relationship interactions, there is no need to list all status if everyone can think it.

File: resources/dicts/thoughts/alive/alive_outside/medicine cat.json
Comment (Lixxis) [inline resources/dicts/thoughts/alive/alive_outside/medicine cat.json:1]: Can we name it medicine_cat.json ? 
Then it would be in line with other naming conventions of the files (and sometimes spaces in file names, lead to problem when using command lines)

File: resources/dicts/thoughts/alive/deputy.json [hunk 3]
```diff
@@ -0,0 +1,69 @@
+[
+    {
+        "id": "deputy_dead_deputy",
+        "thoughts": [
+            "Wishes they could speak to r_c to learn how to assign patrols better"
+        ],
+        "random_status_constraint": [
+            "deputy"
+        ]
```
Comment (Lixxis) [inline resources/dicts/thoughts/alive/deputy.json:9]: Isn't there a 'live status' check?
I think there are missing some conditions, the main cat should also be a deputy, but the 'main_status_constraint' is missing

File: resources/dicts/thoughts/alive/deputy.json
Comment (Lixxis) [inline resources/dicts/thoughts/alive/deputy.json:1]: In general there is missing the 'main_status_constraint' not sure if this is intended.
If so, maybe also remove the constraint of the other files? Just to have a clear structure through all the thought json files.

File: resources/dicts/thoughts/alive/elder.json
Comment (Lixxis) [inline resources/dicts/thoughts/alive/elder.json:1]: Same as alive/deputy json, 'main_status_constraint' is missing. 
If intended, pls remove it from all thoughts json.

File: resources/dicts/thoughts/alive/general.json
Comment (Lixxis) [inline resources/dicts/thoughts/alive/general.json:1]: Don't know if it is implemented as the interaction/relationship stuff, but if so there is no need to list all 'main_stauts_constraint'.
Please check if the list is really needed.

File: resources/dicts/thoughts/alive/kitten.json
Comment (Lixxis) [inline resources/dicts/thoughts/alive/kitten.json:1]: Same as alive/deputy and alive/elder json, 'main_status_constraint' is missing.
If intended, pls remove it from all thoughts json.

File: resources/dicts/thoughts/alive/leader.json
Comment (Lixxis) [inline resources/dicts/thoughts/alive/leader.json:1]: Same as other alive/status json, 'main_status_constraint' is missing.
If intended, pls remove it from all thoughts json.

File: resources/dicts/thoughts/alive/mediator.json [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+            "Is mediating a long time friendship between two Clanmates",
+            "Is feeling frustrated after a failed negotiation",
+            "Is feeling exhausted from resolving petty arguments",
+            "Wants to solve a dispute with another Clan",
+            "Wonders if mediating is meant to be so draining",
+            "Wonders if being a warrior would have been more fun",
+            "Is proud of how well they are doing at mediating",
+            "Accidentally made an argument worse",
+            "Is proud of getting through a difficult mediation session"
+        ]
```
Comment (Lixxis) [inline resources/dicts/thoughts/alive/mediator.json:73]: 'main_status_constraint' is missing

File: resources/dicts/thoughts/alive/medicine cat.json
Comment (Lixxis) [inline resources/dicts/thoughts/alive/medicine cat.json:1]: Can we name it medicine_cat.json ?
Then it would be in line with other naming conventions of the files (and sometimes spaces in file names, lead to problem when using command lines)

and 'main_status_constraint' is missing (see other comments I mentioned it)

File: resources/dicts/thoughts/alive/newborn.json [hunk 5]
```diff
@@ -0,0 +1,31 @@
+[
+    {
+        "id": "gen_newborn",
+        "thoughts": [
+            "Mewls pitifully for milk",
+            "Tries to figure out all of the smells around them",
+            "Snoozes peacefully",
+            "Tries to open their eyes, but it's all blurry"
+        ],
+        "main_age_constraint": [
```
Comment (Lixxis) [inline resources/dicts/thoughts/alive/newborn.json:10]: Is it really main_age_constraint ? (can be)
But everywhere else the 'main_status_constraint' is used, maybe use this constraint to be more 'in line' with the overall structure?

File: resources/dicts/thoughts/alive/warrior.json
Comment (Lixxis) [inline resources/dicts/thoughts/alive/warrior.json:1]: Sometimes 'main_status_constraint' is missing (start at line 162).
See other comments like this.

File: resources/dicts/thoughts/dead/starclan/deputy.json [hunk 6]
```diff
@@ -0,0 +1,11 @@
+[
+    {
+        "id": "deputy_dead_deputy",
+        "thoughts": [
+            "Wishes they could speak to r_c to learn how to assign patrols better"
```
Comment (Lixxis) [inline resources/dicts/thoughts/dead/starclan/deputy.json:5]: A dead cat is thinking this? --> duplicate in another deputy thought (maybe check on that)

File: resources/dicts/thoughts/dead/starclan/deputy.json [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+        ],
+        "random_status_constraint": [
```
Comment (Lixxis) [inline resources/dicts/thoughts/dead/starclan/deputy.json:7]: 'main_status_constaint' is missing (see other comments)

File: resources/dicts/thoughts/dead/starclan/kitten.json [hunk 8]
```diff
@@ -0,0 +1,11 @@
+[
+    {
+        "id": "kit_dead_kit",
+        "thoughts": [
+            "Is wondering if r_c would have been their friend"
+        ],
+        "random_status_constraint": [
+            "kitten"
+        ]
```
Comment (Lixxis) [inline resources/dicts/thoughts/dead/starclan/kitten.json:9]: 'main_status_constaint' is missing (see other comments)

File: resources/dicts/thoughts/dead/starclan/medicine cat.json [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+            "Recieved an ominous and dark message from r_c",
+            "Has been hearing the voices of Dark Forest cats...",
+            "Debates whether or not to ask r_c for gudiance, knowing where they are"
+        ],
+        "main_status_constraint": [
+            "medicine Cat",
+            "medicine Cat Apprentice"
+        ],
+        "random_living_status": [
+            "darkforest"
```
Comment (Lixxis) [inline resources/dicts/thoughts/dead/starclan/medicine cat.json]: why is this thought than in the dead/**starclan**/medicine cat.json

File: resources/dicts/thoughts/dead/starclan/apprentice.json [hunk 10]
```diff
@@ -0,0 +1,23 @@
+[
+    {
+        "id": "apprentice_dead_apprentice",
+        "thoughts": [
+            "Is wondering if r_c would have been their friend"
+        ],
+        "random_status_constraint": [
```
Comment (Lixxis) [inline resources/dicts/thoughts/dead/starclan/apprentice.json:7]: 'main_status_constaint' is missing, yes age will also do, but for all other this constaint is used. Maybe also use this constraint to stay more 'in line' with the style.

File: resources/dicts/thoughts/dead/starclan/medicine cat.json
Comment (Lixxis) [inline resources/dicts/thoughts/dead/starclan/medicine cat.json:1]: Can we name it medicine_cat.json ?
Then it would be in line with other naming conventions of the files (and sometimes spaces in file names, lead to problem when using command lines)

File: resources/dicts/thoughts/dead/starclan/medicine cat.json [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+        ],
+        "main_status_constraint": [
+            "medicine Cat",
+            "medicine Cat Apprentice"
+        ],  
+        "random_status_constraint": [
+            "Kitten"
+        ],
+        "random_living_status": [
+            "darkforest"
```
Comment (Lixxis) [inline resources/dicts/thoughts/dead/starclan/medicine cat.json]: why is this thought than in the dead/**starclan**/medicine cat.json

File: scripts/cat/cats.py [hunk 12]
```diff
@@ -33,10 +32,10 @@
 
 from scripts.utility import get_med_cats, get_personality_compatibility, event_text_adjust, update_sprite
 from scripts.game_structure.game_essentials import game, screen
-from scripts.cat.thoughts import get_thoughts
 from scripts.cat_relations.relationship import Relationship
 from scripts.game_structure import image_cache
 from scripts.event_class import Single_Event
+from .thoughts import Thoughts
```
Comment (Lixxis) [inline scripts/cat/cats.py:38]:
```suggestion
from scripts.cat.thoughts import Thoughts
```

File: scripts/cat/cats.py [hunk 13]
```diff
@@ -1030,23 +1029,27 @@ def thoughts(self):
         """ Generates a thought for the cat, which displays on their profile. """
         all_cats = self.all_cats
         other_cat = random.choice(list(all_cats.keys()))
+        game_mode = game.switches['game_mode']
 
         # get other cat
         i = 0
         while other_cat == self.ID and len(all_cats) > 1 or (
-                all_cats.get(other_cat).status in ['kittypet', 'rogue', 'loner']):
+                all_cats.get(other_cat).status in ['kittypet', 'rogue', 'loner', 'former Clancat']):
```
Comment (Lixxis) [inline scripts/cat/cats.py:1037]: Is 'former Clancat' a valid status? 
I thought 'exiled' is used for ex-Clancats.

File: scripts/cat/cats.py [hunk 14]
```diff
@@ -1030,23 +1029,27 @@ def thoughts(self):
...
             other_cat = random.choice(list(all_cats.keys()))
+            # makes sure that a cat won't think about a cat that was dead longer than they've been alive (with 4 moons difference)
+            if all_cats[other_cat].dead and not self.dead and not self.status in ['leader', 'medicine cat']:
+                if (int(all_cats[other_cat].dead_for) + 4) > self.moons:
```
Comment (Lixxis) [inline scripts/cat/cats.py]: I would change it to the way I told you with the relationship stuff.
This would fix the error with the old loner joining a clan fresh and suddenly know freshly died clan cats.

```suggestion
                if all_cats[other_cat].ID not in self.relationships:
```

File: scripts/events.py [hunk 15]
```diff
@@ -551,7 +551,7 @@ def handle_lost_cats_return(self):
         lost_cat = None
         for cat in Cat.outside_cats.values():
             if cat.outside and cat.status not in [
-                    'kittypet', 'loner', 'rogue', 'former clancat'
+                    'kittypet', 'loner', 'rogue', 'former Clancat'
```
Comment (Lixxis) [inline scripts/events.py]: Is 'former Clancat' a valid status?
I thought 'exiled' is used for ex-Clancats.

File: scripts/events.py [hunk 16]
```diff
@@ -628,7 +628,7 @@ def one_moon_outside_cat(self, cat):
         if cat.moons == 1:
             cat.age = "kitten"
             if cat.status not in [
-                'kittypet', 'loner', 'rogue', 'former clancat'
+                'kittypet', 'loner', 'rogue', 'former Clancat'
```
Comment (Lixxis) [inline scripts/events.py]: Is 'former Clancat' a valid status?
I thought 'exiled' is used for ex-Clancats.

File: scripts/screens/cat_screens.py [hunk 17]
```diff
@@ -126,7 +126,7 @@ def bs_blurb_text(cat):
     if backstory != None and backstory in backstory_text:
         return backstory_text.get(backstory, "")
     else:
-        if cat.status in ['kittypet', 'loner', 'rogue', 'former clancat']:
+        if cat.status in ['kittypet', 'loner', 'rogue', 'former Clancat']:
```
Comment (Lixxis) [inline scripts/screens/cat_screens.py]: Is 'former Clancat' a valid status?
I thought 'exiled' is used for ex-Clancats.

File: scripts/screens/cat_screens.py [hunk 18]
```diff
@@ -915,7 +915,7 @@ def generate_column2(self, the_cat):
 
         # STATUS
         if the_cat.outside and not the_cat.exiled and not the_cat.status in ['kittypet', 'loner', 'rogue',
-                                                                             'former clancat']:
+                                                                             'former Clancat']:
```
Comment (Lixxis) [inline scripts/screens/cat_screens.py]: Is 'former Clancat' a valid status?
I thought 'exiled' is used for ex-Clancats.

File: scripts/utility.py [hunk 19]
```diff
@@ -403,25 +403,25 @@ def create_outside_cat(Cat, status, backstory):
...
-            status = "former clancat"
+        elif backstory in Cat.backstory_categories["former_clancat_backstories"]:
+            status = "former Clancat"
         if status == 'kittypet':
             name = choice(names.names_dict["loner_names"])
         elif status in ['loner', 'rogue']:
             name = choice(names.names_dict["loner_names"] +
                                  names.names_dict["normal_prefixes"])
-        elif status == 'former clancat':
+        elif status == 'former Clancat':
```
Comment (Lixxis) [inline scripts/utility.py:416]: Is 'former Clancat' a valid status?
I thought 'exiled' is used for ex-Clancats.

File: resources/dicts/thoughts/dead/unknownresidence/medicine cat.json
Comment (Lixxis) [inline resources/dicts/thoughts/dead/unknownresidence/medicine cat.json:1]: Can we name it medicine_cat.json ?
Then it would be in line with other naming conventions of the files (and sometimes spaces in file names, lead to problem when using command lines)

File: resources/dicts/thoughts/dead/darkforest/medicine cat.json
Comment (Lixxis) [inline resources/dicts/thoughts/dead/darkforest/medicine cat.json:1]: Can we name it medicine_cat.json ?
Then it would be in line with other naming conventions of the files (and sometimes spaces in file names, lead to problem when using command lines)

**Review by Lixxis — CHANGES_REQUESTED** [review]

File: resources/dicts/thoughts/dead/darkforest/newborn.json [hunk 20]
```diff
@@ -0,0 +1,31 @@
...
+            "Tries to open their eyes, but it's all blurry"
+        ],
+        "main_age_constraint": [
+            "newborn"
+        ]
+    },
+    {
+        "id": "gen_newborn_to_parent",
+        "thoughts":[
+            "Snuggles closer to r_c",
```
Comment (Lixxis) [inline resources/dicts/thoughts/dead/darkforest/newborn.json:17]: How can a dead newborn snuggle to their parents belly?

Overall these thoughts are for alive newborns and not dead  ones

File: resources/dicts/thoughts/dead/starclan/medicine_cat.json [hunk 21]
```diff
@@ -0,0 +1,59 @@
+[
+    {
+        "id": "med_cat_dead_SC_gen",
+        "thoughts": [
+            "Wishes they could visit r_c in StarClan",
```
Comment (Lixxis) [inline resources/dicts/thoughts/dead/starclan/medicine_cat.json]: Isn't the main cat also in star clan or did I miss something?

General some thoughts follow this pattern.

File: resources/dicts/thoughts/dead/starclan/newborn.json [hunk 22]
```diff
@@ -0,0 +1,31 @@
...
+            "Tries to open their eyes, but it's all blurry"
+        ],
+        "main_age_constraint": [
+            "newborn"
+        ]
+    },
+    {
+        "id": "gen_newborn_to_parent",
+        "thoughts":[
+            "Snuggles closer to r_c",
```
Comment (Lixxis) [inline resources/dicts/thoughts/dead/starclan/newborn.json:17]: Is it on purpose, that the sarclan newborn is snuggling up to an alive cat?

File: resources/dicts/thoughts/dead/unknownresidence/newborn.json [hunk 23]
```diff
@@ -0,0 +1,31 @@
...
+            "Tries to open their eyes, but it's all blurry"
+        ],
+        "main_age_constraint": [
+            "newborn"
+        ]
+    },
+    {
+        "id": "gen_newborn_to_parent",
+        "thoughts":[
+            "Snuggles closer to r_c",
```
Comment (Lixxis) [inline resources/dicts/thoughts/dead/unknownresidence/newborn.json:17]: Same as in starclan. Dead newborn snuggles to alive cat?

**Review by Lixxis — CHANGES_REQUESTED** [review]

File: scripts/game_structure/load_cat.py [hunk 24]
```diff
@@ -282,7 +282,8 @@ def json_load():
 
         # Add faded children
         cat.children.extend(cat.faded_offspring)
-
+        
+        game.switches['error_message'] = f'There was an error when thoughts for cat #{cat} are created.'
```
Comment (lillithrosepup) [inline scripts/game_structure/load_cat.py:286]: why is this here? it looks like it would cause issued on loading cats?
